internal/informer/bbp/informer.go

```diff
@@ -105,14 +105,14 @@ func (i *Informer) updateHandler(oldObj, newObj interface{}) {
 
 	previous, ok := oldObj.(*tfv1alpha2.Terraform)
 	if !ok {
-		i.log.Info("received object is not a Terraform object")
+		i.log.Info("previous object is not a Terraform object", "object", oldObj)
```

Comment:
Is this expected? (I don't know the guts of informers that well). If not, this ought to be a warning or error log, or even a crash.

Reply:
It happens, that's why the log is there. I don't know why, but right now we did not care about it much because until this code is just a filler until we we have to work with it (like we have a terrafrom object and the plan runs with it).

Reply:
That sounds suspicious -- what do you see as the oldObj when you run it?

Reply:
```
{"level":"info","ts":"2023-06-15T12:40:21.432+0200","logger":"informer","msg":"previous object is not a Terraform object","version":"","sha":"","object":{"apiVersion":"infra.contrib.fluxcd.io/v1alpha2","kind":"Terraform","namespace":"flux-system","name":"helloworld"}}
```

It shouldn't throw an error, but after this the state is `ok == false && previous == nil`.

So yes sus, but we will deal with it [soon](https://github.com/weaveworks/tf-controller/issues/664).

Reply:
That was one of my thoughts about it, it does not give me the object, it gives me the reference only. The example, I used to build up this, used it as a kubernetes object, so I assumed it is a kubernetes object. It is still possible we only see a serialized version of the object and that does not contains all information, at the end `oldObj` and `newObj` are only an `interface{}`.